scripts/fetch_istio_releases.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+RELEASE_DIR=${1}
+
+echo ${RELEASE_DIR}
+
+if [ -z "${RELEASE_DIR}" ]; then
+  echo "No directory given"
+  exit 1
+fi
+
+if [ -d "${RELEASE_DIR}" ] ; then
+  cd ${RELEASE_DIR}
+
+  dir=$(pwd)
+
+  curl https://api.github.com/repos/istio/istio/releases | jq -r '.[] | select(.tag_name | test("^[0-9.]+$")) | .assets[] | select(.name | test("linux")) | .browser_download_url' | grep -v "sha256" | grep -v "arm" | grep -v "istioctl" | xargs -n 1 wget
```

Comment:
The releases api is paged and this script just uses the first page (https://docs.github.com/en/rest/reference/repos#releases). Are we sure that it includes all necessary istio releases we want to allow upgrades from? Maybe we need to specify a hardcoded list with versions instead.

Reply:
Originally I was concerned that hard coding the versions would limit us on OOB releases and back supporting a version, but as I think through that, we should still be fine if the list is hard coded. I like the idea of not having to hard code because it minimizes mistakes. 

As it stands now, the first page of the api has back to 1.8.4. Currently 1.8 is EOL and 1.9 is about to be EOL, so with that in mind I would argue that it is in fact all of the "necessary" istio releases BUT that does not cover all of the istio versions we have released. 

So it just depends on our chart support model - @cbron any thoughts on this - I know we discussed this a little already.

Reply:
After talking with Caleb, it has been determined that OOB releases will causes issues regardless of if we get all of the versions from the first few pages of the api or if we use a hard coded list. 

Here is the OOB scenario that will cause issues: 
```
Rancher 2.5: we support 1.9.5,1.9.6
Rancher 2.6: we support 1.9.5,1.9.6 and 1.10.4

Then istio upstream releases 1.9.8 for security issues and we do an OOB release for it. 
Because we already released 1.10.4, there will be no air-gap support to upgrade from 1.9.8 to 1.10.4
```

The solution to this will be: 
1. We will specify a list of hard coded versions that will be bundled with the installer image. 
2. We will provide a volume mount that can be used optionally for a user to supply their own tgz that nginx will serve up for the upgrade command - this will catch all edge cases of possible new versions that don't exist in the installer image.

Reply:
Its not about airgap as much that the manifests for 1.9.8 won't be built into 1.10.4 no matter what we do. I think hardcoding a list of all of our old istio releases up to this point is a better idea, even though it sucks we have to manually update it every time. We can add a comment to the version in dockerfile to add this version to the list.